src/hooks/useCampaignManagement.ts

```diff
@@ -0,0 +1,289 @@
+import { useCallback, useState } from "react";
+import { useEventEmitter, EVENT_TYPES } from "../lib/event-bus";
+import type { CampaignEvent } from "../lib/event-bus";
+import {
+  authenticatedFetchWithExpiration,
+  getStoredJwt,
+} from "../services/auth-service";
+import { API_CONFIG } from "../shared";
+import type { Campaign } from "../types/campaign";
+
+interface UseCampaignManagementProps {
+  isAuthenticated: boolean;
+  onSendNotification?: (message: string) => void;
+}
+
+export function useCampaignManagement({
+  isAuthenticated,
+  onSendNotification,
+}: UseCampaignManagementProps) {
+  const [campaigns, setCampaigns] = useState<Campaign[]>([]);
+  const [campaignsLoading, setCampaignsLoading] = useState(false);
+  const [campaignsError, setCampaignsError] = useState<string | null>(null);
+  const [campaignName, setCampaignName] = useState("");
+  const [campaignDescription, setCampaignDescription] = useState("");
+
+  const emit = useEventEmitter();
+
+  // Fetch campaigns
+  const fetchCampaigns = useCallback(async () => {
+    if (!isAuthenticated) return;
+
+    try {
+      setCampaignsLoading(true);
+      setCampaignsError(null);
+
+      const jwt = getStoredJwt();
+      if (!jwt) {
+        setCampaignsError("No authentication token available");
+        return;
+      }
+
+      const response = await authenticatedFetchWithExpiration(
+        API_CONFIG.buildUrl(API_CONFIG.ENDPOINTS.CAMPAIGNS.BASE),
+        { jwt }
+      );
+
+      if (!response.response.ok) {
+        throw new Error(
+          `Failed to fetch campaigns: ${response.response.status}`
+        );
+      }
+
+      const data = (await response.response.json()) as {
+        campaigns: Campaign[];
+      };
+      setCampaigns(data.campaigns || []);
+    } catch (error) {
+      console.error("Failed to fetch campaigns:", error);
+      setCampaignsError(
+        error instanceof Error ? error.message : "Failed to fetch campaigns"
+      );
+    } finally {
+      setCampaignsLoading(false);
+    }
+  }, [isAuthenticated]);
+
+  const handleCreateCampaign = useCallback(async () => {
+    if (!campaignName.trim()) return;
```

Comment:
<img width="444" height="174" alt="Screenshot 2025-09-07 at 15 26 20" src="https://github.com/user-attachments/assets/13be160b-3765-4ec0-a6a4-af9384eba828" />

you have several checks like these, none of them are type safe :(